scripts/privacy_scan_review.py

```
@@ -0,0 +1,384 @@
+        or any(component in {"", ".", ".."} for component in path.split("/"))
+        or any(PATH_COMPONENT.fullmatch(component) is None for component in path.split("/"))
+    ):
+        raise ReviewRecordError(f"owner-review record {label} is not canonical")
+    return path
+
+
+def _secure_read(path: Path) -> tuple[bytes, int]:
+    no_follow = getattr(os, "O_NOFOLLOW", None)
+    nonblocking = getattr(os, "O_NONBLOCK", None)
+    if no_follow is None or nonblocking is None:
+        raise ReviewRecordError("safe owner-review file reads are unavailable")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)